pubsubs/kafka.py

```diff
@@ -0,0 +1,78 @@
+""" Concrete kafka implementation."""
+import logging
+
+from interface_meta import override
+
+from pubsubs.base import MessageQueue, Subscriber
+
+
+class KafkaClient(MessageQueue):
+    """ Concrete client implementation of Kafka."""
+
+    BACKENDS = ["kafka"]
+
+    @override
+    def _connect(self):
+        from confluent_kafka import Producer
+
+        self._prepare()
+        self._producer = Producer(self._config)
+
+    def _prepare(self):
+        """ Prepare kafka configuration."""
+        # Configuration for publisher
+        self._config = self.config.pop("publisher")
+        self._poll = self._config.pop("poll")
+
+        servers = ",".join(self.listeners)
+        self._config.update({"bootstrap.servers": servers})
+
+        # Configuration for subscriber
+        self._subscriber_config = self.config.pop("subscriber")
+
+    @override
+    def _publish(self, topic, message):
+        """ Kafka implementation of publish."""
+
+        def delivery_report(err, msg):
+            if err is not None:
+                logging.exception("Message failed: {}".format(err))
+            else:
+                logging.debug(
+                    "Message delivered, topic: {}, partition: {}, offset".format(
+                        msg.topic(), msg.partition(), msg.offset()
+                    )
+                )
+
+        self._producer.poll(self._poll)
+        self._producer.produce(topic, message, callback=delivery_report)
+        self._producer.flush()
```

Comment:
Do we wanna `flush` every time? A `poll` will retrieve the call back anyway, `flush` is usually used (for example) in a `finally` clause.

Reply:
As per https://github.com/confluentinc/confluent-kafka-python, flushing should be called outside of the data stream context (which for us is... never).
If it is possible to `flush` when the application hits an exception and the connection to the publisher is still open, this would be ideal.

Reply:
I need you help on this one